scripts/check_if_test_should_run_v2.ts

```diff
@@ -39,33 +53,64 @@ async function checkIfTestShouldRun() {
     logMessage('No commits found; this is most likely a bug in the checking script')
     process.exit(1)
   }
-  for (const commit of branchCommits) {
-    logMessage(`\nChecking commit ${commit}...`)
-    const paths: string[] = dirs.concat(['../../yarn.lock'])
-    const anyPathsChanged: boolean = await checkIfAnyPathsChangedInCommit(commit, paths)
-    if (anyPathsChanged) {
+
+  const circleciChangeCommit = await getChangeCommit(join(rootDirectory, '.circleci', 'config.yml'))
+  const yarnLockChangeCommit = await getChangeCommit(join(rootDirectory, 'yarn.lock'))
+  if (
+    branchCommits.includes(yarnLockChangeCommit) ||
+    branchCommits.includes(circleciChangeCommit)
+  ) {
+    // always run tests when yarn.lock or circlici config have changed
+    console.info('true')
+    return
+  }
+
+  const changedPackages = await getChangedPackages(branchCommits)
+  logMessage(`Found ${changedPackages.length} changed packages (${changedPackages.join(', ')})`)
+
+  function hasChangedDependencies(packageName: string): Boolean {
+    if (changedPackages.includes(packageName)) {
+      return true
+    }
+
+    for (const dep of dependencyGraph[packageName]) {
+      return hasChangedDependencies(dep)
```

Comment:
Does this work? It seems like for a package A that depends on packages B and C, `hasChangedDependencies(A)` would return `false` if C changed but B did not.

Perhaps instead you could do `dependencyGraph[packageName].map(hasChangedDependencies).some(identity)` or just `or` every return value together